demo/src/main/java/net/minestom/demo/PlayerInit.java

```diff
@@ -47,131 +47,15 @@
-                            Component.text("Welcome!"),
-                            FrameType.TASK,
-                            Material.IRON_SWORD
-                    );
-                    NotificationCenter.send(notification, event.getPlayer());
-                }
-            })
-            .addListener(PlayerPacketOutEvent.class, event -> {
-                //System.out.println("out " + event.getPacket().getClass().getSimpleName());
-            })
-            .addListener(PlayerPacketEvent.class, event -> {
-                //System.out.println("in " + event.getPacket().getClass().getSimpleName());
-            })
-            .addListener(PlayerUseItemOnBlockEvent.class, event -> {
-                if (event.getHand() != Player.Hand.MAIN) return;
-
-                var itemStack = event.getItemStack();
-                var block = event.getInstance().getBlock(event.getPosition());
-
-                event.getPlayer().sendMessage("MESSAGE " + ThreadLocalRandom.current().nextDouble());
-
-                if ("false" .equals(block.getProperty("waterlogged")) && itemStack.material().equals(Material.WATER_BUCKET)) {
-                    block = block.withProperty("waterlogged", "true");
-                    System.out.println("SET WATERLOGGER");
-                } else if ("true" .equals(block.getProperty("waterlogged")) && itemStack.material().equals(Material.BUCKET)) {
-                    block = block.withProperty("waterlogged", "false");
-                    System.out.println("SET NOT WATERLOGGED");
-                } else return;
-
-                event.getInstance().setBlock(event.getPosition(), block);
-
-            })
-            .addListener(PlayerBlockPlaceEvent.class, event -> {
-//                event.setDoBlockUpdates(false);
-            })
-            .addListener(PlayerBlockInteractEvent.class, event -> {
-                var block = event.getBlock();
-                var rawOpenProp = block.getProperty("open");
-                if (rawOpenProp == null) return;
+    private final ServerFacade serverFacade;
 
-                block = block.withProperty("open", String.valueOf(!Boolean.parseBoolean(rawOpenProp)));
-                event.getInstance().setBlock(event.getBlockPosition(), block);
-            });
-
-    static {
-        InstanceManager instanceManager = MinecraftServer.getInstanceManager();
+    public PlayerInit(ServerFacade serverFacade) {
+        this.serverFacade = serverFacade;
+        InstanceManager instanceManager = serverFacade.getInstanceManager();
 
         InstanceContainer instanceContainer = instanceManager.createInstanceContainer(DimensionType.OVERWORLD);
         instanceContainer.setGenerator(unit -> unit.modifier().fillHeight(0, 40, Block.STONE));
-        instanceContainer.setChunkSupplier(LightingChunk::new);
+        instanceContainer.setChunkSupplier((instance, chunkX, chunkZ) -> new LightingChunk(serverFacade, instance, chunkX, chunkZ));
```

Comment:
The `server` from the Instance should be passed into the chunk supplier, this should remain a method reference.

Reply:
resolved